src/Api/Models/Response/CollectionResponseModel.cs

```diff
@@ -26,19 +26,74 @@ public CollectionResponseModel(Collection collection, string obj = "collection")
     public string ExternalId { get; set; }
 }
 
+/// <summary>
+/// Response model for a collection that includes a user's permission when that user is assigned to the collection.
+/// </summary>
 public class CollectionDetailsResponseModel : CollectionResponseModel
 {
-    public CollectionDetailsResponseModel(CollectionDetails collectionDetails)
+    /// <summary>
+    /// Create a response model for when the user is not assigned to the collection
+    /// (they have no explicit permissions/assignment).
+    /// </summary>
+    private CollectionDetailsResponseModel(Collection collection)
+        : base(collection, "collectionDetails")
+    {
+        ReadOnly = false;
+        HidePasswords = false;
+        Manage = false;
+        Assigned = false;
+    }
+
+    /// <summary>
+    /// Create a response model for when the user is assumed to be assigned to the collection with permissions.
+    /// e.g. The collection details comes from a repository method that only returns collections the user is assigned to.
+    /// </summary>
+    private CollectionDetailsResponseModel(CollectionDetails collectionDetails)
         : base(collectionDetails, "collectionDetails")
     {
         ReadOnly = collectionDetails.ReadOnly;
         HidePasswords = collectionDetails.HidePasswords;
         Manage = collectionDetails.Manage;
+        Assigned = true;
+    }
+
+    /// <summary>
+    /// Create a response model for when a user may or may not be assigned to the collection with permissions.
+    /// </summary>
+    public CollectionDetailsResponseModel(CollectionAdminDetails collection)
+        : base(collection, "collectionDetails")
+    {
+        ReadOnly = collection.ReadOnly;
+        HidePasswords = collection.HidePasswords;
+        Manage = collection.Manage;
+        Assigned = collection.Assigned;
     }
 
     public bool ReadOnly { get; set; }
     public bool HidePasswords { get; set; }
     public bool Manage { get; set; }
+
+    /// <summary>
+    /// Flag for whether the user has been explicitly assigned to the collection either directly or through a group.
+    /// </summary>
+    public bool Assigned { get; set; }
```

Comment:
I think the `Assigned` property is starting to collapse the different classes here. So far you've kept the old & new uses separate: `CollectionDetails` is for user vault data (always assigned) and `CollectionAdminDetails` is for admin actions (maybe assigned). I expected to see a flow like:

* CollectionDetails -> CollectionDetailsResponseModel
* CollectionAdminDetails -> CollectionAdminDetailsResponseModel (which includes optional group/member access details)

but where it would not be possible to do, e.g. CollectionDetails -> CollectionAdminDetailsResponseModel.

The benefit of this is that the type system enforces the separate uses, rather than being careful about calling `FromAssigned` or `FromUnassigned`. And if you're using `FromUnassigned`, then you have a response class where 4 properties are effectively unused (`assigned` bool + the 3 permissions), which suggests a different use of classes & subclasses.

That said, I know this has been tricky to integrate with existing code so this may be a compromise or there may be other uses I'm missing.

Reply:
Very good points, thanks for bringing this up. Approaching it this way allowed me to clean up the response models quite a bit and make it easier to understand. 

Removed the `FromAssigned` and `FromUnassigned` static constructors and the `Assigned` property from `CollectionDetails` (as it is already implied as assigned). Commit [859ab5c](https://github.com/bitwarden/server/pull/3793/commits/859ab5cb97f7a5f11090d11a7edfc2b18102f93d)

Reply:
Looks much better, nice work.